src/Symfony/Component/Security/Core/User/UserProviderInterface.php

```diff
@@ -49,6 +52,8 @@ public function refreshUser(UserInterface $user);
     /**
      * Whether this provider supports the given user class.
      *
+     * @param class-string<TUser> $class
```

Comment:
this is wrong. `supportsClass` will be called with other kind of class strings as well (if should return false for classes that are not TUser, but the way you wrote this would mean that it is forbidden to pass other class strings)

Reply:
Indeed

Reply:
```
@psalm-assert-if-true class-string<TUser> $class
```
seems better then

Reply:
I don't think this assertion would provide much value.